lib/src/spotify.dart

```diff
@@ -7,6 +7,9 @@ class SpotifyApi extends SpotifyApiBase {
   SpotifyApi(SpotifyApiCredentials credentials)
       : super(credentials, http.Client());
 
+  SpotifyApi.withRefreshCallback(SpotifyApiCredentials credentials, Function(SpotifyApiCredentials) callback)
+      : super(credentials, http.Client(), callback);
+
```

Comment:
It's a matter of taste, but I would add the refresh callback as an optional parameter to the default constructor.
And give more indicative name to the parameter, it isn't clear what is the callback for
```dart
SpotifyApi(SpotifyApiCredentials credentials, {Function(SpotifyApiCredentials) onCredentialsRefreshed}) 
```
@hayribakici @stefanoromanello what do you think?

Reply:
@roisnir I agree, having an optional parameter makes it cleaner.

Reply:
Yes you are right. I will commit these changes